### PR TITLE
docs: add documentation for asserts

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 
docs for asserts! should cover changes in these commits:

#9107 
#9099 
#9091 
#9086 
#9066

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 1]
```diff
@@ -851,6 +851,30 @@ assert response.status() == 200;
 The `send()` method adds authentication headers as specified in 
 [ClientOptions](api/io/confluent/ksql/api/client/ClientOptions.html).
 
+Assert the existence of a topic or schema<a name="direct-http-requests"></a>
```
Comment (jnh5y) [inline docs/developer-guide/ksqldb-clients/java-client.md:854]: I'm not great with my Markdown syntax... is the `<a name="direct-http-requests"></a>` stray?

File: docs/reference/sql/appendix.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ The following table shows all keywords in the language.
 | `AND`          | logical "and" operator                  | `WHERE userid<>'User_1' AND userid<>'User_2'`                        |
 | `ARRAY`        | one-indexed array of elements           | `SELECT ARRAY[1, 2] FROM s1 EMIT CHANGES;`                           |
 | `AS`           | alias a column, expression, or type     |                                                                      |
-| `ASSERT`       | assert value, stream, table, tombstone  | `ASSERT NULL VALUES sourceName (columns)? KEY values;`               |
+| `ASSERT`       | assert value, stream, table, tombstone, topic, schema  | `ASSERT NULL VALUES sourceName (columns)? KEY values;`|
```
Comment (jnh5y) [inline docs/reference/sql/appendix.md:22]: Should the optional timeouts be mentioned?  I suppose I'm unclear on the point of this page.  Is it examples or more like an EBNF?

**Review by jnh5y — APPROVED** [review]
LGTM.  We should probably give @JimGalasyn a chance to review as well.

File: docs/reference/sql/appendix.md (hunk 2 above)
Comment (jzaralim) [reply docs/reference/sql/appendix.md:22]: Yeah, thanks for catching that. It's a list of all keywords in ksqlDB's grammar.

File: docs/developer-guide/ksqldb-clients/java-client.md (hunk 1 above)
Comment (jzaralim) [reply docs/developer-guide/ksqldb-clients/java-client.md]: its a bad copy paste job. fixed.

File: docs/reference/sql/appendix.md (hunk 2 above)
Comment (jnh5y) [reply docs/reference/sql/appendix.md:22]: Gotta earn that reviewer cred somehow:).

**Pull request merged**